source/extensions/filters/network/mongo_proxy/proxy.cc

```diff
@@ -176,14 +178,20 @@ void ProxyFilter::decodeQuery(QueryMessagePtr&& message) {
   active_query_list_.emplace_back(std::move(active_query));
 }
 
-void ProxyFilter::chargeQueryStats(const std::string& prefix,
+void ProxyFilter::chargeQueryStats(Stats::StatNameVec& names,
                                    QueryMessageInfo::QueryType query_type) {
-  scope_.counter(fmt::format("{}.query.total", prefix)).inc();
+  names.push_back(mongo_stats_->query_);
+  names.push_back(mongo_stats_->total_);
```

Comment:
do you think it might be worth adding something like `ASSERT(names.capacity() - names.size() >= 2);` as the first line to ensure the caller has reserved the appropriate capacity? The `names.reserve(6)` on line 155 seems a bit mystical without knowing the workings of `chargeQueryStats`, so an `ASSERT` can help enforce a small contract here and prevent a regression if line 155 is removed.

Reply:
good idea.